src/google/adk/runners.py

```diff
@@ -198,6 +199,9 @@ async def run_async(
         if not event.partial:
           await self.session_service.append_event(session=session, event=event)
         yield event
+    finally:
+      # Ensure the span is finished when the function exits
+      span.end()
```

Comment:
with statements still call __exit__ (see https://docs.python.org/3/reference/compound_stmts.html#with) on exceptions. Does this code change here and others to not use with fix anything, or is the entirety of the fix contained in the loop_agent.py file where the final event is yielded on escalation?

Reply:
The reason this fixes anything is because it the code no longer relies on `contextvars` (used under the hook in `start_as_current_span()`) which is where the error comes from. However that has the consequence that any spans created under this function call will not become children of this span.

FYI span is a context manager so this will be equivalent to this change
```py
with tracer.start_span():
  ...
```